src/components/BlogForm/TitleEntry.tsx

```diff
@@ -0,0 +1,112 @@
+"use client";
+import {
+  Button,
+  Container,
+  Stack,
+  TextField,
+  Typography,
+  useTheme,
+} from "@mui/material";
+import { postBlog } from "api/blog";
+import { LoadingScreen } from "components/global-components";
+import _ from "lodash";
+import { useSession } from "next-auth/react";
+import { useSnackbar } from "notistack";
+import { Controller, useForm } from "react-hook-form";
+import { v4 as uuid } from "uuid";
+import { titleDataProps } from "./types";
+
+const TitleEntry = () => {
+  const theme = useTheme();
+  const { enqueueSnackbar } = useSnackbar();
+  const { data: session } = useSession();
+
+  const { control, handleSubmit } = useForm({
+    defaultValues: {
+      title: "",
+      date: "",
+    },
+  });
+
+  if (!session?.user) {
+    return <LoadingScreen />;
+  }
+
+  const { user } = session;
+
+  const onSubmit = (data: titleDataProps) => {
+    const blogSlug = _.kebabCase(data.title);
+    const blogPostId = blogSlug + uuid();
+    postBlog(blogPostId, {
+      ...data,
+      author: user.name,
+      slug: blogSlug,
+      date: new Date().toDateString(),
+    })
+      .then(() => {
+        enqueueSnackbar("Success! A new blog post has been initialised");
```

Comment:
initialised is a little technical, for the snackbars just say if the blog post was successfully created or failed to create... no need to put success of error that can be gained from the styling